### PR TITLE
[Destruction] Legendaries part 2

### PR DESCRIPTION
**Pull request by sMteX**
Not much to add here either, hope it's all fine.

### CODE REVIEW

File: src/Parser/DestructionWarlock/Modules/Items/Legendaries/AlythesssPyrogenics.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  bonusDmg = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasFinger(ITEMS.ALYTHESSS_PYROGENICS.id);
+  }
+
+  on_byPlayer_damage(event) {
+    const enemy = this.enemies.getEntity(event);
+    if (!enemy || !enemy.hasBuff(SPELLS.ALYTHESSS_PYROGENICS_DEBUFF.id, event.timestamp) || AFFECTED_SPELLS.indexOf(event.ability.guid) === -1) {
```
Comment (rp4rk) [inline src/Parser/DestructionWarlock/Modules/Items/Legendaries/AlythesssPyrogenics.js]: This is entirely up to you, but I would use a [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) to hold these spells. This allows you to use `!AFFECTED_SPELLS.has(event.ability.guid)`, which in my opinion is semantically better than `indexOf`.

File: src/Parser/DestructionWarlock/Modules/Items/Legendaries/SindoreiSpite.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import Combatants from 'Parser/Core/Modules/Combatants';
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+import { formatNumber } from 'common/format';
+
+import getDamageBonus from '../../WarlockCore/getDamageBonus';
+
+const SINDOREI_SPITE_DAMAGE_BONUS = .15;
+
+class SoulHarvest extends Module {
```
Comment (rp4rk) [inline src/Parser/DestructionWarlock/Modules/Items/Legendaries/SindoreiSpite.js]: Forgot to change class name?

File: src/Parser/DestructionWarlock/Modules/Items/Legendaries/AlythesssPyrogenics.js (hunk 1 above)
Comment (sMteX) [reply src/Parser/DestructionWarlock/Modules/Items/Legendaries/AlythesssPyrogenics.js]: As discussed on Discord, I'll leave it for now, and after this PR is merged, I'll look if I can rework both Affli and Destro to use Set, it does seem semantically better.

File: src/Parser/DestructionWarlock/Modules/Items/Legendaries/SindoreiSpite.js (hunk 2 above)
Comment (sMteX) [reply src/Parser/DestructionWarlock/Modules/Items/Legendaries/SindoreiSpite.js]: Fixed

**Comment by MartijnHols** [comment]
Thanks!

**Pull request merged**